### PR TITLE
chore(bpmn-model): allow timer start events in model

### LINKED ISSUE #1786: Allow timer start events in XML
Change the model validation to allow start events to also be timers

### PR DESCRIPTION
**Pull request by miguelpires**
Allows BPMN models to specify start events with timer definitions.

closes #1786

### CODE REVIEW

File: bpmn-model/src/main/java/io/zeebe/model/bpmn/validation/zeebe/StartEventValidator.java [hunk 1]
```diff
@@ -15,20 +15,33 @@
...
-    if (!element.getEventDefinitions().isEmpty()) {
-      validationResultCollector.addError(0, "Must be a none start event");
+    final Collection<EventDefinition> eventDefinitions = element.getEventDefinitions();
+    if (eventDefinitions.size() > 1) {
+      validationResultCollector.addError(0, "Start event can't have more than one type");
+    } else {
+      for (EventDefinition eventDef : eventDefinitions) {
+        if (!SUPPORTED_EVENT_TYPE.isAssignableFrom(eventDef.getClass())) {
+          validationResultCollector.addError(
+              0, "Start event must of one of the following types: none, timer");
```
Comment (menski) [inline bpmn-model/src/main/java/io/zeebe/model/bpmn/validation/zeebe/StartEventValidator.java]:
```suggestion
              0, "Start event must be one of the following types: none, timer");
```

**Review by menski — CHANGES_REQUESTED** [review]
A sub process can only have a none start event, at the moment a sub process with a timer would validate, which I think we should prevent.

**Comment by MiguelPires** [comment]
Changed the subprocess validation to only allow none start events

**Review by menski — APPROVED** [review]

**Comment by MiguelPires** [comment]
Sorry, forgot. All done now

**Pull request merged**